starter/src/main/java/com/starter/fullstack/dao/InventoryDAO.java

```diff
@@ -60,32 +66,47 @@ public Inventory create(Inventory inventory) {
 
   /**
    * Retrieve Inventory.
+   *
    * @param id Inventory id to Retrieve.
    * @return Found Inventory.
    */
+
   public Optional<Inventory> retrieve(String id) {
-    // TODO
-    return Optional.empty();
+    Inventory inventory = this.mongoTemplate.findById(id, Inventory.class);
+    return Optional.ofNullable(inventory);
```

Comment:
no need to do this on multiple lines. You can make it 

return Optional.ofNullable(this.mongoTemplate.findById(id, Inventory.class);